pkg/manager/member/tidb_member_manager_test.go

```diff
@@ -2438,3 +2438,16 @@ func mustTiDBConfig(x interface{}) *v1alpha1.TiDBConfigWraper {
 
 	return c
 }
+
+func TestBuildRandomPasswordSecret(t *testing.T) {
+	g := NewGomegaWithT(t)
+	tc := &v1alpha1.TidbCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "basic",
+		},
+	}
+	tmm, _, _, _ := newFakeTiDBMemberManager()
+	_, password := tmm.BuildRandomPasswordSecret(tc)
```

Comment:
Emm, I think this case is unstable because the generated password is random. 
FYI, we can use the lib "github.com/agiledragon/gomonkey/v2" to mock the return for `util.FixedLengthRandomPasswordBytes`.

For example:
```go
patch:= gomonkey.ApplyFunc(util.FixedLengthRandomPasswordBytes, func() []byte {
      return "123123\zxd", nil   // return the invalid string firstly
})
defer patch.Reset()
```

Reply:
If i mock "123123\zxd" password which contain `\` character, the code will block because the loop can't break.  So i use a common password to test.

Reply:
you can add a var to decide password.
```go
        var invalid bool
	patch := gomonkey.ApplyFunc(util.FixedLengthRandomPasswordBytes, func() []byte {
                if invalid {
                  invalid = true
                  return []byte("12\34") // return the invalid string firstly        
                }
		return []byte("1234") // return the invalid string firstly        
	})
```